Palaso/WritingSystems/LdmlDataMapper.cs

```diff
@@ -493,12 +493,37 @@ public void Write(string filePath, WritingSystemDefinition ws, Stream oldFile, W
 					readerSettings.ProhibitDtd = false;
 					reader = XmlReader.Create(oldFile, readerSettings);
 				}
-				using (var writer = XmlWriter.Create(filePath, CanonicalXmlSettings.CreateXmlWriterSettings()))
+				
+				string backupFilePath = null;
+				if (File.Exists(filePath))
 				{
-					writer.WriteStartDocument();
-					WriteLdml(writer, reader, ws);
-					writer.Close();
+					try
+					{
+						backupFilePath = Path.ChangeExtension(Path.GetTempFileName(), Path.GetExtension(filePath));
+						File.Copy(filePath, backupFilePath);
+					}
+					catch (Exception)
+					{
+						backupFilePath = null;
+					}
 				}
+				try
+				{
+					using (var writer = XmlWriter.Create(filePath, CanonicalXmlSettings.CreateXmlWriterSettings()))
+					{
+						writer.WriteStartDocument();
+						WriteLdml(writer, reader, ws);
+					}
+				}
+				catch (Exception)
+				{
+					if (backupFilePath != null)
+						File.Copy(backupFilePath, filePath);
```

Comment:
(Sorry for any confusion about my previous comment which I just accidentally deleted)
Tom, I meant my comment about overwriting on copy to refer to the copy from the backupFilePath to the filePath.
If the XmlWriter crashed while writing it could be leaving an empty or corrupt file at the filePath location and the File.Copy of the backupFilePath would fail and throw an exception in that situation unless you tell it that it can overwrite the file, right?